### PR TITLE
Sanity check ADC pins and FAST_PWM_FAN with Servos

### PR DESCRIPTION
**Pull request by p3p**
Some sanity checks to help with common LPC176x configuration issues

### CODE REVIEW

**Comment by thinkyhead** [comment]
Tweaked this to check more temp pins, if they exist.

**Comment by p3p** [comment]
2 and 3 are valid values for the adc pins

**Comment by thinkyhead** [comment]
> 2 and 3 are valid values for the adc pins

Yeah, I figured just checking against integers is not enough. I can use a macro trick to check if they are set to integer values rather than set to the correct pins format.

**Comment by p3p** [comment]
Checking against the integers I was using is fine for detecting old pins files the format doesn't matter but those invalid values used to the defaults so it's a simple test.

File: Marlin/src/HAL/HAL_LPC1768/inc/SanityCheck.h [hunk 1]
```diff
@@ -28,6 +28,35 @@
   #error "framework-arduino-lpc176x package is out of date, Please update the PlatformIO platforms, frameworks and libraries."
 #endif
 
+/**
+ * Detect an old pins file by checking for old ADC pins values.
+ */
+#define _OLD_TEMP_PIN(P) PIN_EXISTS(TEMP_BED) && _CAT(P,_PIN) <= 7 && _CAT(P,_PIN) != 2 && _CAT(P,_PIN) != 3
```
Comment (LinFor) [inline Marlin/src/HAL/HAL_LPC1768/inc/SanityCheck.h:34]: Seems that "TEMP_BED" typo breaks all builds

**Pull request merged**